### DISCUSSION TITLE
Add GLSL support/syntax highlighting to script editor?

### DISCUSSION THREAD

**Discussion by emgast** [godotengine/godot-proposals#10984] [Engine Core]
The workflow for writing compute shaders is not ideal.

AFAIK in order to work with compute shaders in Godot I have to download an external program, such as VSCode or Notepad++, then find an extension for GLSL since it doesn't seem to be natively supported basically anywhere. Then I write my shader in that program and go back to Godot to see if it compiles.

This is a hassle and has definitely impeded the learning process. Being able to write them in the Godot editor with full syntax highlighting and auto-completion would be a massive QoL improvement.

**Comment by sockeye-d** [comment]
I think this would be an especially good change with the Compositor finally being implemented, since that is entirely compute shaders.

**Comment by Calinou** [comment]
Using the existing shader language highlighter will probably net a good result out of the box, since its syntax is very similar to GLSL. There might need to be a few additions to handle blocks like `#[compute]` or GLSL-specific keywords, but that's about it.

However, this is dependent on https://github.com/godotengine/godot/issues/72281 being implemented first so that it's really usable. Right now, the built-in script editor isn't meant to be used to edit GLSL shaders, only Godot shaders.
